cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java

```diff
@@ -602,6 +602,33 @@ public void testPackageWithEmptyDependenciesToml() throws IOException {
         Assert.assertTrue(projectPath.resolve("Dependencies.toml").toFile().exists());
     }
 
+    @Test(description = "Compile an empty pacakge with compiler plugin")
+    public void testCompileEmptyProjectWithCompilerPlugin() {
+        Path projectPath = this.testResources.resolve("emptyProjectWithCompilerPlugin");
+        System.setProperty("user.dir", projectPath.toString());
+
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false, true, true);
+        new CommandLine(buildCommand).parse();
+        buildCommand.execute();
+
+        Assert.assertTrue(projectPath.resolve("target").resolve("bala").resolve("wso2-compilerPlugin-any-0.1.0.bala")
+                .toFile().exists());
+    }
+
+    @Test(description = "Build an empty package with compiler plugin")
+    public void testBuildEmptyProjectWithCompilerPlugin() throws IOException {
+        Path projectPath = this.testResources.resolve("emptyProjectWithCompilerPlugin");
+        System.setProperty("user.dir", projectPath.toString());
+
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false, true, false);
+        new CommandLine(buildCommand).parse();
+        buildCommand.execute();
+        String buildLog = readOutput(true);
+
+        Assert.assertEquals(buildLog.replaceAll("\r", ""),
+                getOutput("build-empty-project-with-compiler-plugin.txt"));
+    }
+
```

Comment:
Shall we add more test cases to cover the following scenarios for both `bal build` and `bal build -c`:

1. When the package has only non-default modules
2. When the package has root of the non-default module but no source files in
3. When the package has only test files in the default module
4. When the default module is empty but the non-default module has test files

Reply:
Added all requested test cases